### PR TITLE
Relate quote to orders

### PR DESCRIPTION
**Pull request by lexerom**
## Summary

Replace subscriptionId with orderId where orderId relates to Orders. Move order related changes to `orderChanges` property.

[Preview](https://rebilly-website--rem-git-rem-01hnb1ejh7zd3eqe72m-42313a.preview.redocly.app/docs/dev-docs/api/Quotes/)

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by nanobash — DISMISSED** [review]

**Review by nanobash — DISMISSED** [review]

**Review by i-yasko — DISMISSED** [review]

**Comment by lexerom** [comment]
Another idea is to add `orderItemId` or `subscriptionId` to Quote.items.

**Review by Alisa007 — DISMISSED** [review]

**Review by olasunkanmi04 — DISMISSED** [review]

**Review by giokaxo — DISMISSED** [review]

**Comment by lexerom** [comment]
php please review
js please review

**Comment by lexerom** [comment]
js please review

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 1]
```diff
@@ -92,147 +77,181 @@ properties:
...
-          maxLength: 255
-        usageLimits:
-          type:
-            - 'object'
-            - 'null'
-          default: null
-          description: |-
-            Specifies the soft and hard usage limit thresholds for an item with a metered billing plan.
-            This value is ignored when the plan is not metered.
+  order:
```
Comment (justnero) [inline openapi/components/schemas/QuoteChangeOrder.yaml:80]: Am I right this object is the same across all the files? If so, we need it to be extracted and reused as it's quite big and hard to manage otherwise.

Comment (lexerom) [reply openapi/components/schemas/QuoteChangeOrder.yaml:80]: Moved similar code of items into files.

File: openapi/components/schemas/QuoteOrderItem.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    $ref: ./CreatedTime.yaml
+  updatedTime:
+    $ref: ./UpdatedTime.yaml
+  _embedded:
+    type: object
+    description: Embedded objects that are requested by the `expand` query parameter.
+    readOnly: true
+    properties:
+      product:
+        type: object
```
Comment (adamaltman) [inline openapi/components/schemas/QuoteOrderItem.yaml:115]: Why is this here? There is not even a `_links`.

File: openapi/components/schemas/QuoteOrderItem.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
+type: object
+required:
+  - plan
+properties:
+  quoteItemId:
```
Comment (ashkarpetin) [inline openapi/components/schemas/QuoteOrderItem.yaml:5]: Should `id` related to quote item, and `orderItemId` relate to order item instead

File: openapi/components/schemas/QuoteOrderItem.yaml (hunk 2 above)
Comment (lexerom) [reply openapi/components/schemas/QuoteOrderItem.yaml:115]: I think initially it was taken from [SubscriptionOrOneTimeSaleItem.yaml](https://github.com/Rebilly/api-definitions/blob/main/openapi/components/schemas/SubscriptionOrOneTimeSaleItem.yaml)

File: openapi/components/schemas/QuoteOrderItem.yaml (hunk 3 above)
Comment (lexerom) [reply openapi/components/schemas/QuoteOrderItem.yaml:5]: The idea is to eventually replace QuoteOrderItem with OrderSubscription.yaml and OrderOneTimeSaleItem.yaml. But yeah, not 100% sure here, it was `id` for quote item initially.

File: openapi/components/schemas/QuoteOrderItem.yaml (hunk 2 above)
Comment (lexerom) [reply openapi/components/schemas/QuoteOrderItem.yaml]: Removed embedded.

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 4]
```diff
@@ -95,124 +77,43 @@ properties:
...
+    description: Related order properties.
+    properties:
+      id:
+        description: ID of the related order.
+        readOnly: true
+        type:
+          - 'string'
+          - 'null'
+        maxLength: 50
+        example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/QuoteCreateOrder.yaml:91]: It will be nice and consistent to have the same identifiers for each resource.

```suggestion
        example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 5]
```diff
@@ -92,147 +77,31 @@ properties:
...
+          $ref: ./QuoteChangeOrderItem.yaml
+      autopay:
+        type: boolean
+        default: false
+        description: |-
+          Specifies if payment attempts for the related subscription are made automatically.
+          If autopay is enabled, the payment is retrieved from the customer on the renewal date using the payment instrument that is set at `paymentInstrumentId`,
+          or the default payment instrument on the subscription.
+      shipping:
+        $ref: ./Shipping.yaml
```
Comment (vykhrystiuk) [inline openapi/components/schemas/QuoteChangeOrder.yaml:104]: Don't we need billing and delivery address here?

File: openapi/components/schemas/QuoteCreateOrder.yaml (hunk 4 above)
Comment (vykhrystiuk) [reply openapi/components/schemas/QuoteCreateOrder.yaml]: I think there are a couple of more places we should change if we want it to be consistent

File: openapi/components/schemas/QuoteChangeOrder.yaml (hunk 5 above)
Comment (lexerom) [reply openapi/components/schemas/QuoteChangeOrder.yaml:104]: Right now change quotes do not support that. It is basically a proxy for change items. Can be added later when we support that.

**Review by creocoder — DISMISSED** [review]

**Review by creocoder — DISMISSED** [review]

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 6]
```diff
@@ -92,147 +78,30 @@ properties:
...
-    type: boolean
-    default: false
-  interimOnly:
-    type: boolean
-    default: true
-    description: |-
-      Specifies if the quotation invoice must include interim items only.
-      If this value is set to `false`, all upcoming items are included.
+    maxLength: 50
+    example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/QuoteChangeOrder.yaml:85]: To unify the same prefix example for each resource.

```suggestion
    example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

File: openapi/components/schemas/QuoteOrderItem.yaml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    description: ID of the quote item.
+    readOnly: true
+    type: string
+    example: qti_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/QuoteOrderItem.yaml:9]: Please fix the incorrect prefix `qti` to `qt_itm`.

```suggestion
    example: qt_itm_01HXCEQNR3F1G2A6RX6HPS3KFY
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 8]
```diff
@@ -86,147 +71,31 @@ properties:
...
-    type: boolean
-    default: false
-  interimOnly:
-    type: boolean
-    default: true
-    description: |-
-      Specifies if the quotation invoice must include interim items only.
-      If this value is set to `false`, all upcoming items are included.
+    maxLength: 50
+    example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:79]:
```suggestion
    example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 9]
```diff
@@ -88,124 +71,43 @@ properties:
...
-      - $ref: ./ContactObject.yaml
-      - type: 'null'
+  orderId:
+    description: ID of the related order.
+    readOnly: true
+    type:
+      - 'string'
+      - 'null'
+    maxLength: 50
+    example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:81]:
```suggestion
    example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 10]
```diff
@@ -86,147 +71,41 @@ properties:
...
-    type:
-      - 'string'
-      - 'null'
+  orderId:
+    description: ID of the related order.
+    readOnly: true
+    type: string
     maxLength: 50
-    example: 4f6cf35x-2c4y-483z-a0a9-158621f77a21
+    example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:79]:
```suggestion
    example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

File: openapi/components/schemas/StorefrontQuoteTrialConversionOrder.yaml [hunk 11]
```diff
@@ -78,122 +63,39 @@ properties:
...
-          readOnly: true
-          properties:
-            product:
-              type: object
+  orderId:
+    description: ID of the related order.
+    readOnly: true
+    type: string
+    maxLength: 50
+    example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/StorefrontQuoteTrialConversionOrder.yaml:79]:
```suggestion
    example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

File: openapi/components/schemas/QuoteOrderItem.yaml [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+  id:
+    description: ID of the related order item.
+    type: string
+    example: sub_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/QuoteOrderItem.yaml:13]: Order item could be subscription or one-time-sale, thus will be better to remove ambiguous example here.

```suggestion
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 13]
```diff
@@ -92,147 +78,40 @@ properties:
...
-      If this field is omitted, the subscription payment instrument remains unchanged.
-    type:
-      - 'string'
-      - 'null'
+  orderId:
+    description: ID of the related order.
+    type: string
     maxLength: 50
-    example: 4f6cf35x-2c4y-483z-a0a9-158621f77a21
+    example: ord_0YV7DES3WPC5J8JD8QTVNZBZNZ
```
Comment (nanobash) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:85]:
```suggestion
    example: ord_01HVKA5975PJBSQ1SX72G3MSZC
```

**Review by nanobash — DISMISSED** [review]

**Review by noon-dawg — APPROVED** [review]
Looks good on the docs side

**Review by nanobash — APPROVED** [review]

**Review by creocoder — APPROVED** [review]

**Review by justnero — APPROVED** [review]

**Comment by lexerom** [comment]
Moved to monorepo
